### DISCUSSION TITLE
yieldable-json

### DISCUSSION THREAD

**Discussion by Droogans** [DefinitelyTyped/DefinitelyTyped#55269] [Request a new @types package]
https://github.com/ibmruntimes/yieldable-json/pulls?q=is%3Apr+type+definitions
https://www.typescriptlang.org/dt/search?search=yieldable-json

```
$> npm install --save-dev @types/yieldable-json
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@types%2fyieldable-json - Not found
npm ERR! 404
npm ERR! 404  '@types/yieldable-json@*' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
```

I am not an experienced contributor on either of these projects but I decided to make a mention of it here since I'll have to write up just enough of a type interface to test it out in my project.

**Comment by Droogans** [comment]
This is one of those projects that's just between doing too little by writing typing only, and just rewriting the thing. It's good javascript but there's a few things that I think can come out of it by using more of the newer features in the language.

https://gist.github.com/Droogans/88a150817886d6dba827aa74fe26747d

This only implements parsing currently. I am still working on better types, and some tests to prove it isn't broken but it at least runs. Can't be much worse than native sync parse calls.

**Comment by martin-badin** [comment]
Done #57566
